src/planscape/forsys/raster_condition_fetcher.py

```diff
@@ -51,87 +73,107 @@ class RasterConditionFetcher:
 
-    def __init__(self, region: str, priorities: list[str], geo: GEOSGeometry):
+    def __init__(
+            self, region: str, priorities: list[str],
+            land_attributes: list[str],
+            geo: GEOSGeometry):
         raster_geo = get_raster_geo(geo)
 
         conditions = get_conditions(region, priorities)
-        self.conditions_to_raster_values = \
-            self._fetch_conditions_to_raster_values(
-                conditions, raster_geo)
+        attributes = get_attributes(land_attributes)
+        self.raster_values = \
+            self._fetch_raster_values(conditions, attributes, raster_geo)
 
         self.topleft_coords = self._get_topleft_coords(
-            self.conditions_to_raster_values)
+            self.raster_values)
         self.data, self.x_to_y_to_index = self._reformat_to_dataframe(
-            self.conditions_to_raster_values, priorities)
+            self.raster_values, self.topleft_coords, priorities, land_attributes)
         self.width, self.height = self._get_width_and_height(
             self.x_to_y_to_index)
-        
-        # TODO: Only stands with at least one condition value are saved; 
-        # however, some stands may have 0 condition values, but still be useful 
-        # to include in the forsys input (think stand threshold, EPW). These 
+
+        # TODO: Only stands with at least one condition value are saved;
+        # however, some stands may have 0 condition values, but still be useful
+        # to include in the forsys input (think stand threshold, EPW). These
         # should be added to self.data.
 
+    def _fetch_raster_values(self,
+                             conditions, attributes, raster_geo) -> dict[str, RasterPixelValues]:
+        raster_values = {}
+        self._fetch_condition_raster_values(
+            conditions, raster_geo, raster_values)
+        self._fetch_attribute_raster_values(
+            attributes, raster_geo, raster_values)
+        return raster_values
+
     # Fetches condition raster values for a given GEOSGeometry and emits it in
     # a {condition name: ConditionPixelValues} dictionary.
-    def _fetch_conditions_to_raster_values(
+    def _fetch_condition_raster_values(
             self, conditions: list[Condition],
-            geo: GEOSGeometry) -> dict[str, ConditionPixelValues]:
-        conditions_to_raster_values = {}
+            geo: GEOSGeometry, raster_values: dict[str, RasterPixelValues]):
         for c in conditions:
             name = c.condition_dataset.condition_name
             values = get_condition_values_from_raster(geo, c.raster_name)
             if values is None:
                 raise Exception(
                     "plan has no intersection with condition raster, %s" %
                     name)
-            conditions_to_raster_values[name] = values
-        return conditions_to_raster_values
+            raster_values[name] = values
+
+    def _fetch_attribute_raster_values(
```

Comment:
Worth adding documentation similar to _fetch_raster_values

Reply:
Done. Added documentation, and fixed function flow.